src/core/change_detection/constants.ts

```diff
@@ -0,0 +1,91 @@
+import { isBlank } from '../../facade/lang';
+
+/**
+ * Describes the current state of the change detector.
+ */
+export enum ChangeDetectorState {
+  /**
+   * `NeverChecked` means that the change detector has not been checked yet, and
+   * initialization methods should be called during detection.
+   */
+  NeverChecked,
+
+  /**
+   * `CheckedBefore` means that the change detector has successfully completed at least
+   * one detection previously.
+   */
+  CheckedBefore,
+
+  /**
+   * `Errored` means that the change detector encountered an error checking a binding
+   * or calling a directive lifecycle method and is now in an inconsistent state. Change
+   * detectors in this state will no longer detect changes.
+   */
+  Errored
+}
+
+
+/**
+ * Describes within the change detector which strategy will be used the next time change
+ * detection is triggered.
+ */
+export enum ChangeDetectionStrategy {
```

Comment:
using const enums you can reduce extra generated code, see simple example

http://bit.ly/1VE9lw7

only constrain is that you can't use const enum by dynamically accessing it's properties like so:

```typescript
ChangeDetectionStrategy[ someVariable ]
```